src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionFixProvider.cs

```diff
@@ -0,0 +1,175 @@
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        var progress = new StreamingProgressCollector();
+        await SymbolFinder.FindReferencesAsync(
+            symbol, document.Project.Solution, progress, documents: null,
+            FindReferencesSearchOptions.Default, cancellationToken).ConfigureAwait(false);
+
+        return progress.GetReferencedSymbols();
+    }
+
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        foreach (var diagnostic in diagnostics)
+        {
+            await FixOneAsync(editor, document, diagnostic, cancellationToken).ConfigureAwait(false);
+        }
+
+        var changedRoot = editor.GetChangedRoot();
+
+        // Get the language specific rule for formatting this construct and call into the
+        // formatter to explicitly format things. Note: all we will format is the new
+        // node as that's the only node that has the appropriate annotation on it.
+        var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(cancellationToken).ConfigureAwait(false);
+        var rules = ImmutableArray.Create(MultiLineConditionalExpressionFormattingRule.Instance);
+        var spansToFormat = FormattingExtensions.GetAnnotatedSpans(changedRoot, new());
+
+        var formattedRoot = CSharpSyntaxFormatting.Instance.GetFormattingResult(changedRoot, spansToFormat, formattingOptions, rules, cancellationToken).GetFormattedRoot(cancellationToken);
+        changedRoot = formattedRoot;
+
+        editor.ReplaceNode(editor.OriginalRoot, changedRoot);
+    }
+
+    private static async Task FixOneAsync(
+        SyntaxEditor editor, Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+    {
+        var throwNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+        var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+        Contract.ThrowIfNull(methodOrProperty);
+        Contract.ThrowIfFalse(methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax);
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var memberSymbol = semanticModel.GetRequiredDeclaredSymbol(methodOrProperty, cancellationToken);
+        var references = await FindReferencesAsync(document, memberSymbol, cancellationToken).ConfigureAwait(false);
+        var copilotService = document.GetRequiredLanguageService<ICopilotCodeAnalysisService>();
+
+        var (implementationSuggestion, isQuotaExceeded) = await copilotService.ImplementNotImplementedExceptionAsync(
+            document, throwNode.Span, methodOrProperty, memberSymbol, semanticModel, references, cancellationToken).ConfigureAwait(false);
```

Comment:
why pass the document and the semantic model?  the receiver can get the semantic model from the doc.